pkg/trustedcabundle/trustedcabundle.go

```diff
@@ -168,54 +167,41 @@ func ConfigureTrustedCABundle(ctx context.Context, cli client.Client, log logr.L
-		return err
-	}
-
-	for i := range namespaceList.Items {
-		ns := &namespaceList.Items[i]
-		// check namespace status if not Active, then skip
-		if ns.Status.Phase != corev1.NamespaceActive {
-			continue
-		}
-
-		if ShouldInjectTrustedBundle(ns) {
-			if err := wait.PollUntilContextTimeout(ctx, time.Second*1, time.Second*10, false, func(ctx context.Context) (bool, error) {
+// when DSCI TrustedCABundle.ManagementState is set to `Managed`, add new configmap into all active namespaces.
+func AddCABundleCMInAllNamespaces(ctx context.Context, cli client.Client, log logr.Logger, dscInit *dsciv1.DSCInitialization) error {
+	var multiErr *multierror.Error
+	processErr := cluster.ProcessAllNamespace(ctx, cli, func(ns *corev1.Namespace) error {
+		if ShouldInjectTrustedBundle(ns) { // only work on namespace that meet requirements and status active
+			pollErr := wait.PollUntilContextTimeout(ctx, time.Second*1, time.Second*10, false, func(ctx context.Context) (bool, error) {
 				if cmErr := CreateOdhTrustedCABundleConfigMap(ctx, cli, ns.Name, dscInit.Spec.TrustedCABundle.CustomCABundle); cmErr != nil {
 					// Logging the error for debugging
-					fmt.Printf("error creating cert configmap in namespace %v: %v", ns.Name, cmErr)
+					log.Info("error creating cert configmap in namespace", "namespace", ns.Name, "error", cmErr)
 					return false, nil
 				}
 				return true, nil
-			}); err != nil {
-				return err
-			}
+			})
+			multiErr = multierror.Append(multiErr, pollErr)
 		}
+		return nil // Always return nil to continue processing
+	})
+	if processErr != nil {
+		return processErr
 	}
-	return nil
+	return multiErr.ErrorOrNil()
 }
 
-// when DSCI TrustedCABundle.ManagementState is set to `Removed`.
-func RemoveCABundleConfigMapInAllNamespaces(ctx context.Context, cli client.Client) error {
+// when DSCI TrustedCABundle.ManagementState is set to `Removed`, delete configmap from all active namespaces.
+func RemoveCABundleCMInAllNamespaces(ctx context.Context, cli client.Client) error {
 	var multiErr *multierror.Error
-
-	namespaceList := &corev1.NamespaceList{}
-	if err := cli.List(ctx, namespaceList); err != nil {
-		return multierror.Append(multiErr, err)
-	}
-
-	for i := range namespaceList.Items {
-		ns := &namespaceList.Items[i]
-		// check namespace status if not Active, then skip
-		if ns.Status.Phase != corev1.NamespaceActive {
-			continue
-		}
-		if err := DeleteOdhTrustedCABundleConfigMap(ctx, cli, ns.Name); err != nil {
-			multiErr = multierror.Append(multiErr, err)
+	processErr := cluster.ProcessAllNamespace(ctx, cli, func(ns *corev1.Namespace) error {
+		if !ShouldInjectTrustedBundle(ns) { // skip deletion if namespace does not match critieria
+			return nil
 		}
+		multiErr = multierror.Append(multiErr, DeleteOdhTrustedCABundleConfigMap(ctx, cli, ns.Name))
+		return nil // Always return nil to continue processing
+	})
+	if processErr != nil {
+		return processErr
 	}
 	return multiErr.ErrorOrNil()
```

Comment:
Dont we want to return all the errors? We are still swallowing collected `multiErr` when the `processErr` occurs.

```suggestion
	return multiErr.Append(multiErr, processErr).ErrorOrNil()
```

should do the trick here.